### PR TITLE
Optional torch

### PR DESCRIPTION
**Pull request by rflperry**
#### Reference Issues/PRs
#69 

#### What does this implement/fix? Explain your changes.
Modifies installation to make torch (and dependencies for torch related files) optional

#### Any other comments?
Documentation for installation updated as well. See below

image

image

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -1,55 +1,69 @@
...
+
+with optional dependencies
+
 - torch
 - torchvision
 - tqdm
+- pillow
 
 Hardware requirements
 ---------------------
```
Comment (gmischl1) [inline docs/install.rst:69]: Can you put the single quotes around the "mvlearn" under Testing?

File: docs/install.rst [hunk 2]
```diff
@@ -1,55 +1,69 @@
...
-Install the current release of ``mvlearn`` with ``pip``::
+Install the current release of ``mvlearn`` with ``pip3``::
 
-    $ pip install mvlearn
+    $ pip3 install mvlearn
 
 To upgrade to a newer release use the ``--upgrade`` flag::
 
-    $ pip install --upgrade mvlearn
+    $ pip3 install --upgrade mvlearn
```
Comment (gmischl1) [inline docs/install.rst:24]: How do you upgrade if you want the torch dependencies? Does this single command do it, or do you need something like 

pip install --upgrade mvlearn[torch]

**Comment by gmischl1** [comment]
Just so I know, what happens if:

- Someone already installed mvlearn 0.0.1 which has SplitAE, and now decides to do 
```mvlearn
pip install --upgrade mvlearn
```
does this mean they can still use splitAE from the old version, or does that get removed for them?

- Someone has mvlearn 0.0.2 without torch, and wants to upgrade to 0.0.3 at some point in the future with torch. What command do they run?

**Comment by gmischl1** [comment]
Can you add a line to the Notes section of the docstrings for DCCA and SplitAE that says something like

"this function is only available if mvlearn was installed with the optional torch dependencies" and then links to the Install page?

File: docs/install.rst (hunk 2 above)
Comment (rflperry) [reply docs/install.rst:24]: see below. Tell me if refined documentation is needed.

**Comment by rflperry** [comment]
> Just so I know, what happens if:
> 
> * Someone already installed mvlearn 0.0.1 which has SplitAE, and now decides to do
> 
> ```
> pip install --upgrade mvlearn
> ```
> 
> does this mean they can still use splitAE from the old version, or does that get removed for them?
> 
> * Someone has mvlearn 0.0.2 without torch, and wants to upgrade to 0.0.3 at some point in the future with torch. What command do they run?
> * Someone has mvlearn 0.0.2 without torch, but wants torch. Can they just do
> 
> ```
> pip3 install mvlearn[torch]
> ```

**Comment by rflperry** [comment]
Key points:
```mvlearn
pip3 install mvlearn
```
and
```mvlearn
pip3 install mvlearn[torch]
```
differ (irrespective of an option flags) in that the latter also checks extra dependencies. Without [torch], the command will ignore those extras.

## Case 1:
### Scenario
- Someone already installed mvlearn 0.0.1 with torch and then decides to do
```mvlearn
pip3 install --upgrade mvlearn
```
### Answer
torch dependencies will remain the same as before but everything else will be updated. So, in our current case, SplitAE will still function.

## Case 2:
### Scenario
- Someone has mvlearn 0.0.2 without torch, and wants to upgrade to 0.0.3 at some point in the future with torch. What command do they run?

### Answer
```mvlearn
pip3 install --upgrade mvlearn[torch]
```
## Case 3:
### Scenario
- Someone has mvlearn 0.0.2 without torch, but wants torch. Can they just do 
```mvlearn
pip3 install mvlearn[torch]
```
### Answer
Yes

**Pull request merged**